### PR TITLE
use requestAnimationFrame for animation

### PR DESCRIPTION
**Pull request by dlindenkreuz**
- fixed frame rates through `fps` prop
- remove `timeout` prop

animations now look way smoother!

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -80,20 +83,38 @@ class SpriteAnimator extends Component {
...
 
+  animate (nextFrame, time) {
+    if (!this.prevTime) {
+      this.prevTime = time
+    }
+
+    const delta = time - this.prevTime
+    if (delta < this.interval) {
+      this.animationId = raf(time => this.animate(nextFrame, time))
+      return
```
Comment (dlindenkreuz) [inline src/index.js]: bail out if the animation is faster than the specified framerate, check again in the next animation frame. this starts an inner loop. the outer loop componentDidUpdate -> animate -> setState -> componentDidUpdate -> ... is not affected by this, we just "wait" inside of animate until the target framerate is matched.

File: package.json [hunk 2]
```diff
@@ -52,5 +52,8 @@
     "plugins": [
       "transform-es2015-modules-commonjs"
     ]
+  },
+  "dependencies": {
+    "raf": "^3.2.0"
```
Comment (dlindenkreuz) [inline package.json]: using a polyfill here!

**Comment by jcblw** [comment]
👍 dude awesome!

**Comment by jcblw** [comment]
this and a few other things published to `npm` at `0.3.0`

**Comment by jcblw** [comment]
nvm it's at `1.0.0` just realized the removing `timeout` prop is a breaking change. So to get your changes install that version

**Pull request merged**